### PR TITLE
Overdosing on laughter can cause breathloss and lung damage

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

Yo mama's so fat, when she fell I didn't laugh, but the sidewalk cracked up.

## Why It's Good For The Game

Funny joke bro.  SO FUNNY!

## Changelog

:cl:
add: Adds overdose effects for laughter and superlaughter.  You can now laugh so hard you will run out of breath, hurt your lungs, or die.
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
Please change your title to be a clear description of what the PR is for the sake of the tracker

**Comment by Shadowflame909** [comment]
>take laughter/honk pocket for good moodlet
>Get bad moodlet because you start choking

**Comment by timothymtorres** [comment]
> Please change your title to be a clear description of what the PR is for the sake of the tracker

That's a funny joke

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 1]
```diff
@@ -178,11 +179,19 @@
 	SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "chemical_laughter", /datum/mood_event/chemical_laughter)
 	..()
 
+/datum/reagent/consumable/laughter/overdose_process(mob/living/carbon/M, delta_time, times_fired)
+	if(DT_PROB(10, delta_time))
+		to_chat(M, span_userdanger("You laugh so hard you lose your breath..."))
+		M.adjustOxyLoss(5)
+		M.emote("gasp")
+	..()
```
Comment (JohnFulpWillard) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:187]:
```suggestion
	return ..()
```

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 2]
```diff
@@ -178,11 +179,19 @@
 	SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "chemical_laughter", /datum/mood_event/chemical_laughter)
 	..()
 
+/datum/reagent/consumable/laughter/overdose_process(mob/living/carbon/M, delta_time, times_fired)
```
Comment (JohnFulpWillard) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:182]: better var name please

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 3]
```diff
@@ -193,6 +202,12 @@
 		SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "chemical_laughter", /datum/mood_event/chemical_superlaughter)
 	..()
 
+/datum/reagent/consumable/superlaughter/overdose_process(mob/living/carbon/M, delta_time, times_fired)
```
Comment (JohnFulpWillard) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:205]: better var name please

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 4]
```diff
@@ -193,6 +202,12 @@
...
+	if(DT_PROB(16, delta_time))
+		to_chat(M, span_userdanger("You laugh so hard you feel your lungs hurt..."))
+		M.adjustOrganLoss(ORGAN_SLOT_LUNGS, 0.3 * REM * delta_time)
+	..()
```
Comment (JohnFulpWillard) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:209]:
```suggestion
	return ..()
```

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 1 above)
Comment (timothymtorres) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm:187]: Is there a particular reason for this?  Most reagent code in this file uses:

```
..()
```

or

```
..()
. = TRUE
```

Why would we want to return the parent object?

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 2 above)
Comment (timothymtorres) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm]: Fixed with 5b9202127ee67e0c0e5ece15d0c81a157349c9eb

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 3 above)
Comment (timothymtorres) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm]: Fixed with 5b9202127ee67e0c0e5ece15d0c81a157349c9eb

**Comment by timothymtorres** [comment]
Converting this to a draft until I can test this locally in a few days.

**Pull request closed without merging**